cashctrl_ledger/ledger.py

```diff
@@ -47,11 +77,75 @@ def add_price():
         """
         raise NotImplementedError
 
-    def add_vat_code():
+    def add_vat_code(
+        self, code: str, rate: float, account: str, inclusive: bool = True,
+        text: str = ""
+    ):
         """
-        Not implemented yet
+        Adds a new VAT code to the CashCtrl account.
+
+        Parameters:
+            code (str): The VAT code to be added.
+            rate (float): The VAT rate, must be between 0 and 1.
+            account (str): The account identifier to which the VAT is applied.
+            inclusive (bool): Determines whether the VAT is calculated as 'NET' 
+                            (True, default) or 'GROSS' (False).
+            text (str): Additional text or description associated with the VAT code.
         """
-        raise NotImplementedError
+        accounts = self._client.list_accounts()
+        account_map = accounts.set_index('number')['id'].to_dict()
+        payload = {
+            "name": code,
+            "percentage": rate*100,
+            "accountId": account_map.get(account, None),
+            "calcType": "NET" if inclusive else "GROSS",
+            "documentName": text,
+        }
+
+        if payload['accountId'] == None:
+            raise ValueError(f"Account '{account}' does not exist.")
```

Comment:
First assigning 'accountId' with default None, and later checking whether the value is 'None' is a bit twisted.

Here's a suggestion for better readability:
```python
    def add_vat_code(
        self, code: str, rate: float, account: str, inclusive: bool = True,
        text: str = ""
    ):
        """
        Adds a new VAT code to the CashCtrl account.

        Parameters:
            code (str): The VAT code to be added.
            rate (float): The VAT rate, must be between 0 and 1.
            account (str): The account identifier to which the VAT is applied.
            inclusive (bool): Determines whether the VAT is calculated as 'NET'
                            (True, default) or 'GROSS' (False).
            text (str): Additional text or description associated with the VAT code.
        """
        accounts = self._client.list_accounts()
        account_map = accounts.set_index('number')['id'].to_dict()
        if account not in account_map:
            raise ValueError(f"Account '{account}' does not exist.")
        payload = {
            "name": code,
            "percentage": rate*100,
            "accountId": account_map[account],
            "calcType": "NET" if inclusive else "GROSS",
            "documentName": text,
        }
        self._client.post("tax/create.json", data=payload)
```